### DISCUSSION TITLE
Is there a way to retrieve Material You colors from Skip?

### DISCUSSION THREAD

**Discussion by paytondevs** [skiptools/skip#86] [Q&A]
Is it possible to get the Dynamic Colors provided by Material You via Skip/Swift?

**Comment by aabewhite** [comment]
I'm sorry, I'm not exactly sure what you're asking. Are you trying to get the underlying androidx.compose.ui.graphics.Color from a SwiftUI.Color? If so, you can do so like:

let swiftUIColor = ...
#if SKIP // Hide from Swift compiler
let composeColor = swiftUIColor.asColor(opacity: 1.0)
#endif


Please let me know if that doesn't answer your question.

**Reply by paytondevs** [reply]
> I'm sorry, I'm not exactly sure what you're asking. Are you trying to get the underlying androidx.compose.ui.graphics.Color from a SwiftUI.Color? If so, you can do so like:
> 
> 
> 
> let swiftUIColor = ...
> 
> #if SKIP // Hide from Swift compiler
> 
> let composeColor = swiftUIColor.asColor(opacity: 1.0)
> 
> #endif
> 
> 
> 
> 
> 
> Please let me know if that doesn't answer your question.

sorry, i should've been more clear!
is there a way to get the current system Material You color palette?

**Comment by aabewhite** [comment]
Ah, I see. I'm not familiar enough with Material You to give you specifics, but I can say that Skip doesn't limit your ability to access any Android API. You can add dependencies on Android libraries and import and use Android packages. So however you'd access the color palette in an Android app, you can do the same thing in your Skip app.

https://skip.tools/docs/platformcustomization/
https://skip.tools/docs/dependencies/
